platform.js

```diff
@@ -154,6 +154,19 @@ class App {
 
     const dest = path.join(this.originalResourcesDir, 'app.asar')
     debug(`Running asar with the options ${JSON.stringify(asarOptions)}`)
+
+    if (asarOptions.filename) {
+      const src = path.resolve(asarOptions.filename)
+
+      return fs.stat(src)
+        .then(stat => {
+          const target = stat.isFile() ? dest : this.originalResourcesDir
```

Comment:
Could you explain how `target` is determined? Shouldn't target always be `dest`?

Reply:
if the user specified a directory (containing their app.asar and additional resources) fs-extra's `copy` requires the dest to be a directory... if the src is a file, it requires dest to be a file.

Reply:
It's not clear from the docs that you can specify a directory for `asar.filename`. Although even if it did, I would disagree with it because the name of the option makes that choice not obvious. I think the value of `asar.filename` should only be an ASAR file.